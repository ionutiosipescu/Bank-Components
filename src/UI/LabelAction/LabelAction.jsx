import React from "react";
import { LabelActionStyle } from "./LabelAction.style";

function LabelAction({ action, children }) {
  return <LabelActionStyle action={action}>{children}</LabelActionStyle>;
}

export default LabelAction;
