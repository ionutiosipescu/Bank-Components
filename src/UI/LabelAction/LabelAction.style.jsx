import styled from "styled-components";

export const LabelActionStyle = styled.h3`
  color: ${(props) =>
    props.action === "Deposit"
      ? "var(--green)"
      : props.action === "Withdraw"
      ? "var(--red)"
      : "var(--gray-dark)"};
`;
