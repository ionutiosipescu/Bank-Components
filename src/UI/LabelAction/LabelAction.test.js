import React from "react";
import { shallow, mount, render } from "enzyme";
import LabelAction from "./LabelAction";

describe("LabelAction component", () => {
  it("should render correctly with status prop", () => {
    const wrapper = shallow(
      <LabelAction action="Deposit">Test label</LabelAction>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly without status prop", () => {
    const wrapper = shallow(<LabelAction>Test label</LabelAction>);
    expect(wrapper).toMatchSnapshot();
  });
});
