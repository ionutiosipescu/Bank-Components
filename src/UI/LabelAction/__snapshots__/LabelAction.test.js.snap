// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelAction component should render correctly with status prop 1`] = `
<styled.h3
  action="Deposit"
>
  Test label
</styled.h3>
`;

exports[`LabelAction component should render correctly without status prop 1`] = `
<styled.h3>
  Test label
</styled.h3>
`;
